pyro/poutine/lower_poutine.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import, division, print_function
+
+from torch.autograd import Variable
+import pyro  # XXX remove this cyclic dependency
+from pyro.params import _PYRO_PARAM_STORE
+
+from .poutine import Poutine
+
+
+class LowerPoutine(Poutine):
```

Comment:
@neerajprad How similar is this is to an HMC poutine? Can we use `poutine.lower` directly for HMC, where we use `LeapFrog` instead of `infer.MAP`?

Reply:
Discussed with @fritzo. We decided to use ReplayPoutine for HMC where the replayed trace has `requires_grad` set to True for `sample` sites, and the `param` sites detached.